enumerables.rb

```diff
@@ -0,0 +1,132 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_each
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x])
+      x += 1
+    end
+    list
+  end
+
+  def my_each_with_index
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x], x)
+      x += 1
+    end
+    list
+  end
+
+  def my_select
+    result = []
+    return to_enum unless block_given?
+
+    my_each do |x|
+      result << x if yield(x)
+    end
+    result
+  end
+
+  def my_all?(input = nil)
+    my_each do |x|
+      return false if block_given? && !yield(x)
+
+      if !block_given? && input.nil?
+        return false unless x
+      elsif input
+        return false unless input_check(x, input)
+      end
+    end
+    true
+  end
+
+  def my_any?(input = nil)
+    my_each do |x|
+      return true if block_given? && yield(x)
+
+      if !block_given? && input.nil?
+        return true if x
+      elsif !block_given? && input
+        return true if input_check(x, input)
+      end
+    end
+    false
+  end
+
+  def my_none?(input = nil)
+    my_each do |x|
+      return false if block_given? && yield(x)
+
+      if !block_given? && input.nil?
+        return false if x
+      elsif !block_given? && input
+        return false if input_check(x, input)
+      end
+    end
+    true
+  end
+
+  def my_count(input = nil)
+    count = 0
+    my_each do |x|
+      if block_given? && input.nil?
+        count += 1 if yield(x)
+      elsif x && !input.nil?
+        count += 1 if x == input
+      elsif input.is_a?(Integer)
+        count += 1
+      else
+        count = length
+      end
+    end
+    count
+  end
+
+  def my_map
+    result = []
+    my_each do |x|
+      return to_enum unless block_given?
+
+      result << yield(x) || result << proc.call(x) if block_given?
+    end
+    result
+  end
```

Comment:
[OPTIONAL]
- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so I strongly recommend working on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```